### PR TITLE
Enable field alingment

### PR DESCRIPTION
**Pull request by jLopezbarb**
# Proposed changes

Fixes LAKE-49

Enabled field alignment by configuring the golangci.yaml file. Field alignment optimizes the memory layout of our data structures, ensuring that they are stored efficiently in memory. This optimization speeds up memory access by generating code that requires just one instruction for reading and writing a memory location. Without alignment, processors might need to use multiple instructions to access data between addresses that are not aligned with the machine word size.

This also reduces our resulting binary size (Not too much but there is it).

## How to Validate

**Validation Scenario:**
Run the golangciyaml script

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

### CODE REVIEW

**Comment by andreafalzetti** [comment]
@jLopezbarb could you have a look at the conflict on `cmd/deploy/proxy_test.go`?

**Comment by andreafalzetti** [comment]
@jLopezbarb can you double check the unit tests? Some are failing 🙏

File: pkg/model/manifest_friendly_err.go [hunk 1]
```diff
@@ -46,7 +47,7 @@ func getManifestSuggestionRules(manifestSchema interface{}) []*suggest.Rule {
 
 	for structName, structKeywords := range manifestKeys {
 		for _, keyword := range structKeywords {
-			// example: line 5: field contex not found in type model.buildInfoRaw
+			// example: line 5: field context not found in type model.buildInfoRaw
```
Comment (AdrianPedriza) [inline pkg/model/manifest_friendly_err.go:50]:
```suggestion
			// example: line 5: field contex not found in type model.buildInfoRaw
```
I think `contex` was the example of a invalid key

File: pkg/suggest/suggest_test.go [hunk 2]
```diff
@@ -15,17 +15,18 @@ package suggest
...
+	"github.com/stretchr/testify/assert"
 )
 
 func Test_errorSuggestion_suggest(t *testing.T) {
 	alwaysFalse := func(e error) bool { return false }
 	returnSameErr := func(e error) error { return e }
 	emptyRule := NewRule(alwaysFalse, returnSameErr)
 
-	inputError := errors.New("line 4: field contex not found in type model.buildInfoRaw")
+	inputError := errors.New("line 4: field context not found in type model.buildInfoRaw")
```
Comment (AdrianPedriza) [inline pkg/suggest/suggest_test.go:29]:
```suggestion
	inputError := errors.New("line 4: field contex not found in type model.buildInfoRaw")
```

File: pkg/suggest/suggest_test.go [hunk 3]
```diff
@@ -37,17 +38,17 @@ func Test_errorSuggestion_suggest(t *testing.T) {
...
 		expected      string
+		rules         []*Rule
 		expectedError bool
 	}{
 		{
 			name:       "basic Rule",
 			inputError: inputError,
 			rules:      []*Rule{ruleNotFoundInBuildInfoRaw},
-			expected:   "line 4: field contex does not exist in the build section of the Okteto Manifest",
+			expected:   "line 4: field context does not exist in the build section of the Okteto Manifest",
```
Comment (AdrianPedriza) [inline pkg/suggest/suggest_test.go:51]:
```suggestion
			expected:   "line 4: field contex does not exist in the build section of the Okteto Manifest",
```

File: pkg/suggest/suggest_test.go [hunk 4]
```diff
@@ -60,7 +61,7 @@ func Test_errorSuggestion_suggest(t *testing.T) {
 				),
 				ruleNotFoundInBuildInfoRaw,
 			},
-			expected: "line 4: field contex does not exist in the build section of the Okteto Manifest. Did you mean \"context\"?",
+			expected: "line 4: field context does not exist in the build section of the Okteto Manifest. Did you mean \"context\"?",
```
Comment (AdrianPedriza) [inline pkg/suggest/suggest_test.go:64]:
```suggestion
			expected: "line 4: field contex does not exist in the build section of the Okteto Manifest. Did you mean \"context\"?",
```

**Review by AdrianPedriza — APPROVED** [review]

**Review by andreafalzetti — APPROVED** [review]

**Pull request merged**